memtable/write_buffer_manager.cc

```diff
@@ -53,7 +53,44 @@ std::size_t WriteBufferManager::dummy_entries_in_cache_usage() const {
   }
 }
 
+void WriteBufferManager::SetBufferSize(size_t new_size) {
+  if (new_size == 0) {
+    assert(false);
+    return;
+  }
```

Comment:
Cleanup3: based on a comment in the code `Cannot early-exit on !enabled() because SetBufferSize(0) needs to unblock...`, I believe we don't want buffer size to set to 0 in runtime. So I added an explicit check here for both prod and debug build.